src/quorum.rs

```diff
@@ -19,19 +21,42 @@ pub enum VoteResult {
 }
 
 /// Index is a Raft log position.
-#[derive(Default, Clone, Copy)]
+#[derive(Default, Clone, Copy, Eq)]
 pub struct Index {
     pub index: u64,
     pub group_id: u64,
 }
 
+impl PartialEq for Index {
```

Comment:
It doesn't feel right to implement `PartialEq` and its family traits for the `Index`. As the same index with different group id should not be treat as the same `Index`. The comparison rule is limited to context and should not apply universally.

Reply:
Maybe remove Ord and PartialOrd?